janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestClientSetup.java

```diff
@@ -0,0 +1,228 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Create an instance of ES REST {@link org.elasticsearch.client.RestClient} from a JanusGraph
+ * {@link org.janusgraph.diskstorage.configuration.Configuration}.
+ */
+public class RestClientSetup {
+
+    private static final Logger log = LoggerFactory.getLogger(RestClientSetup.class);
+
+    public ElasticSearchClient connect(Configuration config) throws IOException {
+        log.debug("Configuring RestClient");
+
+        final List<HttpHost> hosts = new ArrayList<>();
+        final int defaultPort = config.has(INDEX_PORT) ? config.get(INDEX_PORT) : ElasticSearchIndex.HOST_PORT_DEFAULT;
+        final String httpScheme = config.get(ElasticSearchIndex.SSL_ENABLED).booleanValue() ? "https" : "http";
+        for (String host : config.get(INDEX_HOSTS)) {
+            String[] hostparts = host.split(":");
+            String hostname = hostparts[0];
+            int hostport = defaultPort;
+            if (hostparts.length == 2) hostport = Integer.parseInt(hostparts[1]);
+            log.debug("Configured remote host: {} : {}", hostname, hostport);
+            hosts.add(new HttpHost(hostname, hostport, httpScheme));
+        }
+        RestClient rc = getRestClient(hosts.toArray(new HttpHost[hosts.size()]), config);
+
+        final int scrollKeepAlive = config.get(ElasticSearchIndex.ES_SCROLL_KEEP_ALIVE);
+        Preconditions.checkArgument(scrollKeepAlive >= 1, "Scroll Keep alive should be greater or equals than 1");
+        final RestElasticSearchClient client = getElasticSearchClient(rc, scrollKeepAlive);
+        if (config.has(ElasticSearchIndex.BULK_REFRESH)) {
+            client.setBulkRefresh(config.get(ElasticSearchIndex.BULK_REFRESH));
+        }
+
+        return client;
+    }
+
+    protected RestClient getRestClient(HttpHost[] hosts, Configuration config) {
+        final RestClientBuilder restClientBuilder = getRestClientBuilder(hosts);
+
+        final HttpClientConfigCallback httpClientConfigCallback = getHttpClientConfigCallback(config);
+        if (httpClientConfigCallback != null) {
+            restClientBuilder.setHttpClientConfigCallback(httpClientConfigCallback);
+        }
+
+        final RequestConfigCallback requestConfigCallback = getRequestConfigCallback(config);
+        if (requestConfigCallback != null) {
+            restClientBuilder.setRequestConfigCallback(requestConfigCallback);
+        }
+
+        return restClientBuilder.build();
+    }
+
+    protected RestClientBuilder getRestClientBuilder(HttpHost[] hosts) {
+        return RestClient.builder(hosts);
+    }
+
+    protected RestElasticSearchClient getElasticSearchClient(RestClient rc, int scrollKeepAlive) {
+        return new RestElasticSearchClient(rc, scrollKeepAlive);
+    }
+
+    /**
+     * Returns the callback used the default {@link RequestConfig} being set to
```

Comment:
`"Returns the {@link RequestConfig} callback used to customize the {@link CloseableHttpClient}..."`?

Reply:
rephrased both get...Callback descriptions.